### PR TITLE
Feature/112953 rsvp store definition

### PR DESCRIPTION
**Pull request by paulmskim**
For ticket: https://central.tri.be/issues/112953

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,6 +64,7 @@
 		"redux-saga": "^0.16.0",
 		"redux-thunk": "^2.3.0",
 		"reselect": "^3.0.1",
+		"uniqid": "^5.0.3",
 		"url-loader": "^0.6.2",
 		"uuid": "^3.3.2",
```
Comment (bordoni) [inline package.json]: Are we going to remove `uuid`?

Comment (paulmskim) [reply package.json]: yeah good call, was going to but forgot about it.

File: plugins/tickets/src/modules/data/blocks/rsvp/thunks.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Internal dependencies
+ */
+import { actions } from '@moderntribe/tickets/data/blocks/rsvp';
+
+export const setRSVPHeader = ( data ) => ( dispatch ) => {
```
Comment (mitogh) [inline plugins/tickets/src/modules/data/blocks/rsvp/thunks.js]: We should start adding only sagas instead as is already available.

File: plugins/tickets/src/modules/blocks/rsvp/rsvp-advanced-options/template.js [hunk 3]
```diff
@@ -3,6 +3,7 @@
  */
 import React, { Component, Fragment } from 'react';
 import PropTypes from 'prop-types';
+import uniqid from 'uniqid';
```
Comment (mitogh) [inline plugins/tickets/src/modules/blocks/rsvp/rsvp-advanced-options/template.js]: Wonder if the one provided by `lodash` is not enough for this one

File: plugins/common/src/modules/utils/moment.js [hunk 4]
```diff
@@ -216,6 +216,10 @@ export const toTime = ( date, format = dateUtil.FORMATS.WP.time ) => {
 	return date.format( toFormat( format ) );
 }
 
+export const toTime24Hr = ( date, format = dateUtil.FORMATS.WP.time24Hr ) => {
```
Comment (mitogh) [inline plugins/common/src/modules/utils/moment.js]: Wonder if we can just return directly without the need of the explicit `return` here instead.

File: plugins/tickets/src/modules/data/blocks/rsvp/actions.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+} );
+
+export const setRSVPCapacity = ( capacity ) => ( {
+	type: types.SET_RSVP_CAPACITY,
+	payload: {
+		capacity,
+	},
+} );
+
+export const setRSVPEnableNotGoing = ( enableNotGoing ) => ( {
```
Comment (mitogh) [inline plugins/tickets/src/modules/data/blocks/rsvp/actions.js]: Does this one also handles the `disable` case?

Comment (paulmskim) [reply plugins/tickets/src/modules/data/blocks/rsvp/actions.js]: so this is actually a field called `enableNotGoing`. this action just sets the value, either `true` or `false`.

Comment (mitogh) [reply plugins/tickets/src/modules/data/blocks/rsvp/actions.js]: In that case I think something like `setRSVPNotGoing` makes more sense instead

File: plugins/tickets/src/modules/blocks/rsvp/rsvp-advanced-options/template.js (hunk 3 above)
Comment (paulmskim) [reply plugins/tickets/src/modules/blocks/rsvp/rsvp-advanced-options/template.js]: i suppose, though lodash's uniqueid module only provides incremental numbers. I'd be worried that it won't be unique enough. also uniqid is lighter than lodash's uniqueid.

**Comment by paulmskim** [comment]
@mitogh fixed up your suggestions, would like another pass before merging.

**Pull request merged**